agent/structs/config_entry.go

```diff
@@ -269,6 +268,12 @@ func (e *ServiceConfigEntry) Validate() error {
 		validationErr = multierror.Append(validationErr, fmt.Errorf("invalid value for balance_inbound_connections: %v", e.BalanceInboundConnections))
 	}
 
+	switch e.Protocol {
+	case "", "http", "http2", "grpc", "tcp":
```

Comment:
Should we worry about case sensitiveness in here? 🤔

Reply:
The [normalize](https://github.com/hashicorp/consul/blob/main/agent/structs/config_entry.go#L229) routine called before validation makes sure the protocol field is always lowercase

Reply:
We should probably add a comment at Validation/Normalize call sites to state that the order is important and add a test that would fail if they are called out of order.